db/migrate/20210722172410_devise_create_users.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class DeviseCreateUsers < ActiveRecord::Migration[6.1]
+  def change
+    create_table :users do |t|
+      ## Database authenticatable
+      t.string :firstname
+      t.string :lastname
+      t.string :email,              null: false, default: ""
```

Comment:
`default: ""` option means that in DB we will save empty string if user didn't provide email. We can leave only `null: false` option here. Also, the same comment for `encrypted_password`
```suggestion
      t.string :email,              null: false
```